oval_graph/_xml_parser_test_info.py

```diff
@@ -107,18 +107,37 @@ def _xml_element_to_dict(self, object_, collected_object):
         else:
             result[object_.attrib.get('id')] = "does not exist"
             result[self._get_unique_id_in_dict(
-                object_, result)] = self._get_object_items(object_)
+                object_, result)] = self._get_object_items(object_, collected_object)
         return result
 
-    def _get_object_items(self, object_):
+    def _get_object_items(self, object_, collected_object):
         out = {}
         for element in object_.iterchildren():
             if element.text and element.text.strip():
                 out[self._get_unique_id_in_dict(element, out)] = element.text
             else:
-                out[self._get_unique_id_in_dict(element, out)] = "no value"
+                if self._collected_object_is_not_none_and_not_empty_and_element_contain_var_ref(
+                        element, collected_object):
+                    variable_value = ''
+                    var_id = element.attrib.get('var_ref')
+                    for item in collected_object:
+                        if var_id == item.attrib.get('variable_id'):
+                            variable_value += item.text
+                        elif self._get_key_for_element(item) == 'message':
+                            variable_value = item.text[:item.text.find(
+                                '(') + 1] + var_id + ')'
```

Comment:
This feels clumsy. At least the index computation could be refactored out.

Also, what if there is a different message, with different meaning, perhaps completely unrelated?

Reply:
Messages are sometimes destroyed. The message is limited to 99 characters.